### PR TITLE
Remove redundant declaration of address names

### PR DESCRIPTION
**Pull request by maurelian**
**Description**
The strings used to resolve system contracts in the Address Manager must be correct, but they are declared independently in many places. 

This PR defines an exported object to act as a dictionary of correct names.

### CODE REVIEW

File: packages/contracts/deploy/010-AddressDictator.deploy.ts
Comment (maurelian) [inline packages/contracts/deploy/010-AddressDictator.deploy.ts:70]: I know this diff looks a bit odd: why would you only replace one of the two identical strings?

Ideally the two arguments would have different types. This [SE answer](https://stackoverflow.com/a/45801281/7230551) points at a decent way, but I think this is good enough for now.

File: packages/contracts/src/address-names.ts
Comment (smartcontracts) [inline packages/contracts/src/address-names.ts:10]: Stylistic discussion:

Could we do something like:

```
const names = {
  managed: {
     contracts: {
       OVM_BondManager: ...,
       OVM_L1CrossDomainMessenger: ...
       Proxy__OVM_L1CrossDomainMessenger: ...,
     },
     accounts: {
        ...
     }
  },
  unmanaged: {
    ...
  },
}
```

Note using the same naming format as with the `predeploys`.

**Review by maurelian — COMMENTED** [review]
I've found some places I've missed applying the object member access, so this is not quite ready, but appreciate the quick reviews coming in. :)

File: packages/contracts/src/address-names.ts
Comment (maurelian) [reply packages/contracts/src/address-names.ts]: OK, I did that. I used two commits: 
1. https://github.com/ethereum-optimism/optimism/pull/1695/commits/cfc8b9b4e0277a55459d156a46fd9417a7cd8b48: moves everything into a singled nested object.

2. [`88312e2` (#1695)](https://github.com/ethereum-optimism/optimism/pull/1695/commits/88312e24a7db20276e45dcb2a3b1ecc01d96b2bc): changes the names to use the `predeploy.ts` formatting. 

TBH I'm not a fan of the change in \#2, it mean we still sling around these inconsistent prefixes. But I don't want to bikeshed, so it's included here.

File: packages/contracts/deploy/010-AddressDictator.deploy.ts
Comment (smartcontracts) [reply packages/contracts/deploy/010-AddressDictator.deploy.ts:70]: Actually, when `name` and `contract` are the same, we can leave out `contract` entirely. So you could delete the `contract` parameter here.

File: l2geth/rollup/sync_service.go
Comment (smartcontracts) [inline l2geth/rollup/sync_service.go:321]: Looks like this needs a rebase

**Comment by smartcontracts** [comment]
Looking good, I'd rebase into one commit and should be good to go after approval

File: ops/docker/Dockerfile.integration-tests
Comment (smartcontracts) [inline ops/docker/Dockerfile.integration-tests:7]: I think it would be best to have this diff in a different PR if it's necessary for some reason

File: ops/scripts/deployer.sh
Comment (smartcontracts) [inline ops/scripts/deployer.sh:42]: Same comment as above and in other places in this PR where the CI is modified

**Review by smartcontracts — COMMENTED** [review]
Also note that there are other places in this repository where these names are being used. However, that could make a good first issue or two so I'd recommend not updating those values in this PR.

File: ops/docker/Dockerfile.integration-tests
Comment (maurelian) [reply ops/docker/Dockerfile.integration-tests:7]: Agree, that's my plan, I just pushed this to see if it would fix it on GHA too.

**Review by smartcontracts — APPROVED** [review]
LGTM, nice cleanup.

**Review by tynes — APPROVED** [review]

**Pull request merged**
